### PR TITLE
[WIP] refactor LSF on top of AbstractGridEngineBatchSystem

### PR DESCRIPTION
**Pull request by mr-c**
Inspired by https://github.com/BD2KGenomics/toil/pull/1864 I thought I'd bring LSF into the party

- [ ] Test this @ EBI

### CODE REVIEW

File: src/toil/batchSystems/lsf.py [hunk 1]
```diff
@@ -21,262 +21,133 @@
...
-        """We give parasol a second to catch its breath (in seconds)
+    @classmethod
+    def getWaitDuration(cls):
+        """We give LSF a second to catch its breath (in seconds)
         """
-        #return 0.0
         return 15
 
-    @classmethod
-    def getRescueBatchJobFrequency(cls):
```
Comment (ejacox) [inline src/toil/batchSystems/lsf.py]: Just to note - This function doesn't seem to be used and could probably be removed everywhere. There is a test for it, but LSF isn't run currently for our CI tests.

Comment (ejacox) [reply src/toil/batchSystems/lsf.py]: Nevermind, I see that it is in abstractGridEngineBatchSystem.

**Comment by mr-c** [comment]
Hey @ejacox & @tetron , many of the CWL conformance tests pass with this, but many other fail with "This job was passed a promise that wasn't yet resolved when it ran. " Any hint as to what I did to break that?

**Comment by ejacox** [comment]
@mr-c, the code looks good. Thank you for adding this. I think we need to get this into the tests:
http://jenkins.cgcloud.info/job/toil-pull-requests/3251/testReport/src.toil.test.batchSystems.batchSystemTest/

Does that mean I need to install LSF on the Jenkins slave?

**Comment by mr-c** [comment]
@ejacox Thanks, please don't merge until I fix the failing conformance tests.

Yes, LSF/OpenLava needs adding to the test suite

**Comment by mr-c** [comment]
Re-done in https://github.com/BD2KGenomics/toil/pull/2043

**Pull request closed without merging**